Unity-Technologies/Graphics
# add editor pinning

---
### Purpose of this PR
Implemented editor pinning: updating and storing editor revisions in `_latest_editor_versions.metafile` on a nightly basis (3 types: staging `--fast`, latest_public `--published-only`, latest_internal)
- All jobs (referring to trunk) use the revisions from latest_internal. Editor priming for these is no longer required, because the revision build is kicked off already during the night
- All custom-revision jobs continue using editor priming (same way as before), to keep the resources free while waiting for the editor build

How editor pinning works is written under the yamato docs https://github.com/Unity-Technologies/Graphics/tree/yamato/editor-pin-clean/.yamato/docs#editor-priming-vs-editor-pinning 

---
### Testing status

**Manual Tests**: What did you do?
- Set it up also for 9.x.x to ensure that backporting this is safe

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)
Ran multiple jobs (both custom-revision and target revision on trunk):
- custom-revision ABV: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252F_abv.yml%2523all_project_ci_CUSTOM-REVISION/3262935/job
- custom-revision (bits of nightly not covered in abv): android https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252Funiversal-android-opengles3.yml%2523Universal_Android_OpenGLES3_Standalone_CUSTOM-REVISION/3263012/job && https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252Funiversal-android-vulkan.yml%2523Universal_Android_Vulkan_Standalone_CUSTOM-REVISION/3263014/job
- custom-revision (bits of nightly not covered in abv): smoke tests https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252F_abv.yml%2523all_smoke_tests_CUSTOM-REVISION/3262981/job
- custom-revision (bits of nightly not covered in abv): trunk verification  https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252F_abv.yml%2523trunk_verification_CUSTOM-REVISION/3262992/job
- custom-revision pack and test all [package context]: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin/.yamato%252F_packages.yml%2523all_package_ci_CUSTOM-REVISION/3264699/job
- custom-revision pack and test all [project context]: (IS PART OF ABV) https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin/.yamato%252F_projectcontext.yml%2523all_package_ci_project_CUSTOM-REVISION/3264704/job

- trunk nightly (covers ABV): https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252F_abv.yml%2523all_project_ci_nightly_trunk/3262844/job
- trunk pack and test all [package context]: https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252F_packages.yml%2523all_package_ci_trunk/3263110/job
- trunk pack and test all [project context]: (IS PART OF ABV)

- promote/publish dryrun  https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean/.yamato%252F_preview_publish.yml%2523promote_config_dry/3265309/job/pipeline
 

**Yamato**: (Select your branch):
- current 'clean' target branch https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-clean
- the 'in-progress' target branch https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin
- the 'in-progress' ci branch https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics/tree/yamato%252Feditor-pin-ci


---
### Comments to reviewers
Some of the jobs fail, but by investigating the logs can be seen that these are failing tests and getting the editor has succeeded
